src/CRM/CivixBundle/Resources/views/Code/module.civix.php.php

```diff
@@ -82,10 +82,39 @@ public static function findClass($suffix) {
     return self::CLASS_PREFIX . '_' . str_replace('\\', '_', $suffix);
   }
 
+<?php if (\Civix::checker()->hasUpgrader()) { ?>  /**
+   * @return \CiviMix\Schema\SchemaHelperInterface
+   */
+  public static function schema() {
+    return $GLOBALS['CiviMixSchema5']->getHelper(static::LONG_NAME);
+  }
+<?php } ?>
+
 }
 
 use <?php echo $_namespace ?>_ExtensionUtil as E;
 
+<?php if (\Civix::checker()->hasMixinLibrary() && !\Civix::checker()->coreHasPathload()) { ?>
+($GLOBALS['_PathLoad'][0] ?? require __DIR__ . '/mixin/lib/pathload-0.php');
+<?php } ?>
+<?php if (\Civix::checker()->hasMixinLibrary()) { ?>
+pathload()->addSearchDir(__DIR__ . '/mixin/lib');
+<?php } ?>
+<?php if (\Civix::checker()->hasMixinLibrary('civimix-schema@5')) { ?>
+spl_autoload_register('_<?php echo $mainFile ?>_civix_class_loader', TRUE, TRUE);
+
+function _<?php echo $mainFile ?>_civix_class_loader($class) {
+  // This allows us to tap-in to the installation process (without incurring real file-reads on typical requests).
+<?php $_clPrefix = 'CiviMix\\Schema\\' . \CRM\CivixBundle\Utils\Naming::createCamelName($mainFile) . '\\' ?>
+  if (substr($class, 0, <?php echo strlen($_clPrefix) ?>) === <?php var_export($_clPrefix) ?>) {
```

Comment:
This is really brittle. How about:
```suggestion
  if (str_starts_with($class, <?php var_export($_clPrefix) ?>)) {
```

Reply:
"Really brittle" is subjective:

* Yeah, if we're talking about code that a developer edits by hand, then `substr(str,0,n)`  is more brittle than `str_starts_with()`. (*I definitely prefer using `str_starts_with()`...*)
* But `*.civix.php` is machine-generated glue, so the human-editing isn't a big factor. As generated glue code, its interoperability is a consideration. ("So Bob tried the extension on php7+Civi 5.53; he says it works. Carol on php8+Civi 5.52 also says it works. Then Dave tried it on same versions and it crashed -- because his specific permutation, php7+Civi 5.52, lacks the polyfill...") From compatibility POV, `str_starts_with()` is more brittle than `substr()`.

So I'd ordinarily agree, but in this context it's easier to use the Old Ways.

Reply:
@totten somehow I knew you would say that... so how about a compromise:
```suggestion
  if (strpos($class, <?php var_export($_clPrefix) ?>) === 0) {
```

Reply:
@colemanw Yup, that sounds good.